File: Creational/Builder/GoLang/chicken_meal_builder.go
package main

type ChickenMealBuilder struct {
	meal *Meal
}

func NewChickenMealBuilder() *ChickenMealBuilder {
	return &ChickenMealBuilder{meal: &Meal{}}
}

func (c *ChickenMealBuilder) BuildMainItem() {
	c.meal.MainItem = "Chicken Sandwich"
}

func (c *ChickenMealBuilder) BuildSideItem() {
	c.meal.SideItem = "Nuggets"
}

func (c *ChickenMealBuilder) BuildDrink() {
	c.meal.Drink = "Sprite"
}

func (c *ChickenMealBuilder) BuildToy() {
	c.meal.Toy = "Toy Dinosaur"
}

func (c *ChickenMealBuilder) GetMeal() *Meal {
	return c.meal
}

File: Creational/Builder/C#/ChickenMealBuilder.cs
namespace FastFoodBuilder {
    public class ChickenMealBuilder : IMealBuilder {
        private Meal _meal = new Meal();

        public void BuildMainItem() {
            _meal.MainItem = "Chicken Sandwich";
        }

        public void BuildSideItem() {
            _meal.SideItem = "Nuggets";
        }

        public void BuildDrink() {
            _meal.Drink = "Sprite";
        }

        public void BuildToy() {
            _meal.Toy = "Toy Dinosaur";
        }

        public Meal GetMeal() {
            return _meal;
        }
    }
}
